cmd/zdb/zdb.c

```diff
@@ -375,10 +375,11 @@ livelist_verify(dsl_deadlist_t *dl, void *arg)
  * Check for errors in the livelist entry and discard the intermediary
  * data structures
  */
-/* ARGSUSED */
 static int
 sublivelist_verify_lightweight(void *args, dsl_deadlist_entry_t *dle)
 {
+	(void) args;
```

Comment:
Well, it horribly pollutes prototypes and they grow rightward at an extraordinary pace (which combined with the 80-column limit is, uh, yeah), and "elsewhere" uses comments, not annotations (`__maybe_unused` is primarily (pretty much only) used for local variables, and from the few cases I looked at they're all fed to asserts, which I'll prune separately I guess).

Plus, I strongly prefer the `(void) args;` style, since it's a distinct thing that happens in the code, and not disconnected from't. If you use it later on, then you'll remove it. If there's an `#if`, then you use the variables used there in the `#else`, because annotating the argument means it'll fall into disrepair at best and hide bugs at worst (at least the comments didn't actually do anything, but, well, they were still mostly wrong).

Reply:
Alright, I'm convinced.

Reply:
> If you use it later on, then you'll remove it

Does anything enforce that?

Reply:
only your human eyes, unfortunately, but I peg that at a high conversion rate